api/provider/aws/test_aws/deploy_list_test.go

```diff
@@ -0,0 +1,138 @@
+package test_aws
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeployList(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "dpl_id1",
+			EntityType: "deploy",
+			Key:        "name",
+			Value:      "dpl1",
+		},
+		{
+			EntityID:   "dpl_id1",
+			EntityType: "deploy",
+			Key:        "version",
+			Value:      "1",
+		},
+		{
+			EntityID:   "dpl_id1",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task-definition/l0-test-dpl:1",
+		},
+		{
+			EntityID:   "dpl_id2",
+			EntityType: "deploy",
+			Key:        "name",
+			Value:      "dpl2",
+		},
+		{
+			EntityID:   "dpl_id2",
+			EntityType: "deploy",
+			Key:        "version",
+			Value:      "1",
+		},
+		{
+			EntityID:   "dpl_id2",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task-definition/l0-test-dpl:2",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	listTaskDefinitionFamiliesPagesfn := func(input *ecs.ListTaskDefinitionFamiliesInput,
+		fn func(output *ecs.ListTaskDefinitionFamiliesOutput, lastPage bool) bool) error {
+		taskDefinitionFamilies := []*string{
+			aws.String("l0-test-"),
```

Comment:
I may be mistaken, but I believe the family is the name of the task itself without its revision numbers. So I think this would need to be `l0-test-dpl` in your case. Unless this test is already passing for you?

Reply:
would be `l0-test-dpl_id1` and `l0-test-dpl_id2`